SingularityBase/src/main/java/com/hubspot/mesos/CounterMap.java

```diff
@@ -52,12 +52,7 @@ public String toString() {
   }
 
   public void incr(K key, long amount) {
-    Counter c = map.get(key);
-
-    if (c == null) {
-      c = new Counter();
-      map.put(key, c);
-    }
+    Counter c = map.computeIfAbsent(key, k -> new Counter());
```

Comment:
we are not using the key in order to determine the value here, should use `putIfAbsent` instead of `computeIfAbsent`. Same for the one below